pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -632,9 +632,50 @@ func (m *kubeGenericRuntimeManager) computePodActions(pod *v1.Pod, podStatus *ku
 		changes.KillPod = true
 	}
 
+	if utilfeature.DefaultFeatureGate.Enabled(features.SidecarLifecycle) {
+		if pod.Spec.RestartPolicy != v1.RestartPolicyAlways && changes.KillPod == false {
+			onlySidecars := true
+
+			//Determine if there are any non sidecar containers that are running or need restarting
+			//if there are none, we can kill the remaining sidecars
+			for _, container := range pod.Spec.Containers {
+				if !lifecycleSidecar(container) {
+					containerStatus := podStatus.FindContainerStatusByName(container.Name)
+					if kubecontainer.ShouldContainerBeRestarted(&container, pod, podStatus) || containerStatus.State == kubecontainer.ContainerStateRunning {
+						onlySidecars = false
+					}
+				}
+			}
+
+			//only sidecars are left so terminate them all
+			if onlySidecars {
+				for idx, container := range pod.Spec.Containers {
+					containerStatus := podStatus.FindContainerStatusByName(container.Name)
+					// we don't need to terminate non sidecars or exited sidecars
+					if lifecycleSidecar(container) && containerStatus.State == kubecontainer.ContainerStateRunning {
+						message := " All containers have permanently exited, sidecar container will be killed"
```

Comment:
pls add sidecar name to this msg. in practice it's very helpful in debugging as this message will be written into the events kubelet creates

Reply:
yeah good shout will do